### DISCUSSION TITLE
Interactive button almost done...

### DISCUSSION THREAD

**Discussion by Lu1508** [Monika-After-Story/MonikaModDev#7924] [QA - code and content creation (technical)]
Like the tittle says, I almost finish the submod about "Interactive button" (#7701)

I just need to know how to change the button position when Monika is or not leaning

I tried a lot of ways (ones gave me error and other stayed in the same position), but I still can't figure it out 

Can someone help me, please? I promise this is the last time I bother you (._.") 

```
screen nosedot():
    zorder 200
    if store.mas_calendar.enabled and not store._menu:
        imagebutton:
            idle ("mod_assets/buttons/face/dot.png")
            action Function(renpy.call, "monika_boop")
            xpos "buttposx"
            ypos "buttposy"



init python:
    def buttposy():
        MASPoseMap(
            default=244,
            l_default=290
            )


    def buttposx():
        MASPoseMap(
            default=624,
            l_default=672
            )
```

**Comment by Booplicate** [comment]
There's no simple way to detect leaning because Monika's sprite is a complex composite image.

- One thing you could try is to call `renpy.game.context().images.get_attributes` and check current image name (which may or may not be always correct)
 - Another is to access `store.mas_moni_idle_disp.current_exp` and get current expression (only for idle and may also be incorrect)

Also, you opened a few issues, but you could keep it all in 1 as you're working on the same thing.
